src/interface/sparse_pack.hpp

```diff
@@ -131,8 +131,8 @@ class SparsePack : public SparsePackBase {
     // accessed on device via instance of types in the type list Ts...
     // The pack will be created and accessible on the device
     template <class T>
-    SparsePack GetPack(T *pmd) const {
-      return SparsePack(SparsePackBase::GetPack(pmd, *this));
+    SparsePack GetPack(T *pmd, const std::vector<bool> &include_block = {}) const {
+      return SparsePack(SparsePackBase::GetPack(pmd, *this, include_block));
```

Comment:
Is the `include_block` functionality expected to be used in userspace? Or do you plan to have some mechanism for automatically choosing which blocks to include?

Reply:
user space.  i think that's a lot more flexible for downstream codes.  i don't presume to know how someone may eventually use this, beyond the use case we have.  and for us, there is essentially zero burden to passing this in.

Reply:
Yeah ok fair enough